src/index.ts

```diff
@@ -27,69 +23,55 @@ function validate(options: Options) {
       const e = `'${
           pair.invalid}' is not a valid configuration option. Please use '${
           pair.expected}' instead. This library is using Axios for requests. Please see https://github.com/axios/axios to learn more about the valid request options.`;
-      return new Error(e);
+      throw new Error(e);
     }
   }
-  return null;
 }
 
-export function _buildMetadataAccessor(type: string) {
-  return function metadataAccessor(
-      options: string|Options|Callback, callback?: Callback) {
-    if (!options) {
-      options = {};
-    }
-
-    if (typeof options === 'function') {
-      callback = options;
-      options = {};
-    }
-
-    if (typeof options === 'string') {
-      options = {property: options};
-    }
+async function metadataAccessor(type: string, options?: string|Options) {
+  options = options || {};
+  if (typeof options === 'string') {
+    options = {property: options};
+  }
+  let property = '';
+  if (typeof options === 'object' && options.property) {
+    property = '/' + options.property;
+  }
+  validate(options);
+  const ax = axios.create();
+  rax.attach(ax);
+  const baseOpts = {
+    url: `${BASE_URL}/${type}${property}`,
+    headers: {'Metadata-Flavor': 'Google'},
```

Comment:
nit: the string `Metadata-Flavor` is repeated a few time here and in tests in different cases. Please move it to a (possibly exported) variable (say `header`) and use `header` and `header.toLowerCase()` elsewhere.

Reply:
🌮